### PR TITLE
Setup and documentation files for release

### PR DESCRIPTION
**Pull request by cczhu**
Updates to root directory readme and config files, and additional changelog and license sections to the documentation.  These are separate from the source code and most of the docs, so can just be rebased with master before final release on PyPi.

Currently this branch builds into my bare Python2 and Python3 virtualenvs with `pip install -e`, and tests run successfully.

Things to consider before merging this PR:

- Not sure how to include LICENSE.rst as a block of text in the documentation without inciting an error.
- I can try to get a Zenodo DOI for citation, or registry on the Astrophysics Source Code Library before PyPi release.  If we elect to do these after release, then under `Citation` we should recommend citing the GitHub link directly.
- Maybe we should add our astro e-mail addresses to our profiles.
- This repo might need to be moved out of @mhvk's profile.

### CODE REVIEW

**Comment by cczhu** [comment]
The error in the Travis build comes from the first point above.

**Comment by mhvk** [comment]
Hmm, so it looks like the ordering is
1. Make sure we are happy with the location (is a move easy)?
2. Register with ASCL and perhaps Zenodo to have something to put in CITATION (we have to be a bit careful with wording, since we're GPL);
3. Put on pypi
4. Notify astropy affiliated.

**Comment by mhvk** [comment]
Let's just keep the license as GPL 3.

**Comment by mhvk** [comment]
I quite like the idea of the badges.  For a reference, I guess we have to register it at Zenodo or something similar, so we have a DOI.

File: README.rst [hunk 1]
```diff
@@ -1,12 +1,45 @@
...
 
 For installation and usage instructions, please see the `online documentation 
 <https://baseband.readthedocs.io/>`_.
+
+Project Status
+--------------
+
+.. image:: https://travis-ci.org/mhvk/baseband.svg?branch=master
+
+.. image:: https://coveralls.io/repos/github/mhvk/baseband/badge.svg?branch=master
```
Comment (mhvk) [inline README.rst]: The coverage one doesn't show anything if I just "view" this page - while travis does.

File: setup.cfg [hunk 2]
```diff
@@ -39,16 +39,16 @@ exclude=
...
-author = Marten H. van Kerkwijk
+author = Marten H. van Kerkwijk, Chenchong Zhu
 author_email = mhvk@astro.utoronto.ca
 license = GPLv3
 edit_on_github = True
 github_project = mhvk/baseband
 install_requires = astropy
 # version should be PEP440 compatible (http://www.python.org/dev/peps/pep-0440)
-version = 0.0.dev
+version = 1.0.0rc1
```
Comment (mhvk) [inline setup.cfg]: Probably OK, but check astropy affiliated package instructions

Comment (cczhu) [reply setup.cfg]: I think this will be set by the release instructions.

File: README.rst (hunk 1 above)
Comment (cczhu) [reply README.rst]: I revised the badge, but apparently because it points to master, it won't properly display until this PR's been merged.

**Comment by cczhu** [comment]
Now has correct Astropy requirement.

**Comment by cczhu** [comment]
Some minor new changes, but it's been long enough that maybe it's best you have another look?

**Comment by cczhu** [comment]
In the Astropy docs, the version is not officially set until a release is to be made, so my thought was to do the same.

**Pull request merged**